src/Engine/EmitMSIL/Replication.cs

```diff
@@ -50,7 +50,7 @@ public static IList ReplicationLogic(IEnumerable<MethodBase> mInfos, IList args,
             {
                 result = mi.Invoke(reducedArgs[0], reducedArgs.Skip(1).ToArray());
             }
-
+            //TODO return lists/arrays without extra wrapping in next PR.
             return new[] { result };
```

Comment:
Will this help in some cases (I think you suggested something like this):
```
if (result is IList) return (IList)result;

return new[] { result };
```

Reply:
It prevents adding an extra level of nesting.

Reply:
yes, I think it resolves a minimal set of cases, like this test which is currently failing:
```
 public void IndexIntoRange_WithIdent()
        {
            var dscode = @"
import(""DesignScriptBuiltin.dll"");
a = 0..100..10; //a is an ILIST because of replication?
b = 5;
c=a[b];";
            var ast = ParserUtils.Parse(dscode).Body;
            var output = codeGen.EmitAndExecute(ast);
            Assert.IsNotEmpty(output);
            Assert.AreEqual(40, output.Values.ToList()[2][0]);
        }

```